src/provider/provider_os_memory_linux.c

```diff
@@ -6,13 +6,26 @@
 */
 
 #include <assert.h>
+#include <errno.h>
+#include <limits.h>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/syscall.h>
 #include <unistd.h>
 
-#include "provider_os_memory_internal.h"
 #include <umf/providers/provider_os_memory.h>
 
+#include "provider_os_memory_internal.h"
+#include "utils_log.h"
+
+// maximum value of the off_t type
```

Comment:
you could write a comment why this arbitrary `-999999`

Reply:
Fixed